src/views/RecIM/views/Activity/index.jsx

```diff
@@ -159,17 +161,25 @@ const Activity = () => {
     setAdminAnchorEl(e.currentTarget);
   };
 
-  const handleJoinActivity = async () => {
-    setLoading(true);
-    const profileInfo = await userService.getProfileInfo(profile.AD_Username);
-    const request = {
-      Name: profileInfo.fullName,
-      ActivityID: activityID,
-    };
-    await createTeam(profile.AD_Username, request);
-    setReload(!reload);
-    createSnackbar(`Activity ${activity.Name} has been joined successfully`, 'success');
-    setLoading(false);
+  let handleJoinActivity = async () => {
+    if (user?.Status == 'Pending' || user == null) {
+      setOpenWaiver(true);
+      return;
+    }
+    if (activity.SoloRegistration) {
+      setLoading(true);
+      const profileInfo = await userService.getProfileInfo(profile.AD_Username);
+      const request = {
+        Name: profileInfo.fullName,
+        ActivityID: activityID,
+      };
+      await createTeam(profile.AD_Username, request);
+      setReload(!reload);
```

Comment:
There is a "special" way that react handles states which makes this not always work as you intend it to. Instead, inversions like this should be done with a lambda function passed in like so:
```suggestion
      setReload(r => !r);
```